File: fitlog-ui/src/types/supplement.ts
export interface SupplementDto {
  id: string;
  name: string;
  usageNote?: string;
  dosage?: string;
  timing?: string;
}

export interface CreateSupplementRequest {
  name: string;
  usageNote?: string;
  dosage?: string;
  timing?: string;
}

File: FitLog.Domain/Entities/Supplement.cs
using FitLog.Domain.Common;

namespace FitLog.Domain.Entities;

/// <summary>
/// Supplement (takviye) entity'si.
/// Kullanıcının aldığı takviyelerin takibi için.
/// </summary>
public class Supplement : BaseEntity
{
    public Guid UserId { get; set; }
    public string Name { get; set; } = string.Empty;
    public string? UsageNote { get; set; } // Kullanım notu
    public string? Dosage { get; set; } // Doz bilgisi
    public string? Timing { get; set; } // Örn: "Sabah", "Antrenman Öncesi"
    
    // Navigation Property
    public virtual User User { get; set; } = null!;
}

